python/ray/serve/_private/application_state.py

```diff
@@ -1001,31 +1009,32 @@ def _save_checkpoint_func(
 @ray.remote(num_cpus=0, max_calls=1)
 def build_serve_application(
     import_path: str,
-    config_deployments: List[str],
     code_version: str,
     name: str,
     args: Dict,
-) -> Tuple[List[Dict], Optional[str]]:
+    logging_config: LoggingConfig,
+) -> Tuple[Optional[List[Dict]], Optional[str]]:
     """Import and build a Serve application.
 
     Args:
         import_path: import path to top-level bound deployment.
-        config_deployments: list of deployment names specified in config
-            with deployment override options. This is used to check that
-            all deployments specified in the config are valid.
         code_version: code version inferred from app config. All
             deployment versions are set to this code version.
         name: application name. If specified, application will be deployed
             without removing existing applications.
         args: Arguments to be passed to the application builder.
-        logging_config: The application logging config, if deployment logging
-            config is not set, application logging config will be applied to the
-            deployment logging config.
+        logging_config: the logging config for the build app task.
     Returns:
         Deploy arguments: a list of deployment arguments if application
             was built successfully, otherwise None.
         Error message: a string if an error was raised, otherwise None.
     """
+    configure_component_logger(
+        component_name=f"build_{name}_task",
```

Comment:
My only concern is this component name. Currently in Serve we have `controller`, `proxy`, and `replica`. Can definitely add a new type for this, but given this is free formed, it might be hard for users to find them on the platform. Also, depending on whether we want to show these by default, we might want to follow up with a product change to add a new component name filter in the log viewer UI.

Reply:
synced offline, setting component name to `"controller"` and setting component id to `build_default_pid`. Since the component name is used for filtering logs, creating a new component name for the build task is confusing and hard to find. It makes the most sense to group it under controller logs.

Reply:
sounds like the right approach to me too. we can always iterate